Mention Spatial -> Node3D rename in Introduction docs (#7338)
godotengine/godot-docs: tutorials/3d/introduction_to_3d.rst

Following this reasoning, the 3D engine uses the :ref:`Node3D <class_Node3D>`
node for everything 3D.

> .. note:: Be aware that there was a renaming of "Spatial" Nodes to
>    "Node3D" starting Godot 4. Any Godot 3.x references about "Spatial"
>    Nodes refer to "Node3D" in Godot 4.

Reviewer: ```suggestion
.. note:: Be aware that "Spatial" Nodes are now called "Node3D"
   starting with Godot 4.. Any Godot 3.x references to "Spatial"
   Nodes refer to "Node3D" in Godot 4.
```

Author: Is the double dot after `with Godot 4` intentional?

Reviewer 2: I doubt it :)
